pyntc/devices/ios_device.py

```diff
@@ -227,21 +254,93 @@ def close(self):
             self.native.disconnect()
             self._connected = False
 
-    def config(self, command):
-        self._enter_config()
-        self._send_command(command)
-        self.native.exit_config_mode()
+    def config(self, command, **netmiko_args):
+        """
+        Send config commands to device.
+
+        By default, entering and exiting config mode is handled automatically.
+        To disable entering and exiting config mode, pass `enter_config_mode` and `exit_config_mode` in ``**netmiko_args``.
+        This supports all arguments supported by Netmiko's `send_command_set` method using ``netmiko_args``.
+        This will send all commands in ``command`` regardless of failures, and only raise an Error after sending all commands.
+
+        Args:
+            command (str|list): The command or commands to send to the device.
+            **netmiko_args: Any argument supported by ``netmiko.ConnectHandler.send_command_set``.
+
+        Returns:
+            str: The config session input and ouput from sending ``command``.
+
+        Raises:
+            TypeError: When sending an argument in ``**netmiko_args`` that is not supported.
+            CommandError: When one of the commands reports an error on the device.
+
+        Example:
+            >>> device = IOSDevice(**connection_args)
+            >>> device.config("no service pad")
+            'configure terminal\nEnter configuration commands, one per line.  End with CNTL/Z.\n'
+            'host(config)#no service pad\nhost(config)#end\nhost#'
+            >>> device.config(["interface Gig0/1", "description x-connect"])
+            'configure terminal\nEnter configuration commands, one per line.  End with CNTL/Z.\n'
+            'host(config)#interface Gig0/1\nhost(config-if)#description x-connect\nhost(config-if)#end\nhost#'
+            >>>
+        """
+        netmiko_args["config_commands"] = command
+        try:
+            command_response = self.native.send_config_set(**netmiko_args)
+        except TypeError as err:
+            raise TypeError(f"Netmiko Driver's {err.args[0]}")
+
+        self._check_command_output_for_errors(command, command_response)
+        return command_response
+
+    def config_list(self, commands, **netmiko_args):
```

Comment:
Doesn't `config_list` overlap with `config` since they can both accept a list of commands?

Reply:
There is overlap in that both accept sending a list of commands, but they did in how exceptions are handled, and the return date..

| config method | config_list method |
|-|-|
| All commands are sent before checking for an error | Commands are sent one at a time. Each command's result is checked for an error, and sending commands ceases upon detecting an error |
| Raises CommandError upon detecting an error, which would not be as helpful in determining which commands failed | Raises CommandListError upon detecting an error, which provides the list of commands that were sent, as well as which command failed |
| A string of all output from the device interaction is returned | A list of each command's interaction with the device is returned |

Reply:
I think if we want to consolidate these to a single interface, then we would need to do that across all OS Drivers. I'm not opposed to doing that, but would rather make that an issue to be addressed by other work.